pyof/foundation/basic_types.py

```diff
@@ -412,7 +412,7 @@ class BinaryData(GenericType):
     return the size of the instance using Python's :func:`len`.
     """
 
-    def __init__(self, value=b''):  # noqa
+    def __init__(self, value=None):  # noqa
```

Comment:
That test is different and does not cover this case. We need a new test to make it explicit you need the None behavior and that it works as expected.

Writing tests to other structures is out of the scope of this issue/PR.